locallib.php

```diff
@@ -1850,6 +1850,13 @@ function moodleoverflow_delete_post($post, $deletechildren, $cm, $moodleoverflow
             // Delete the read records.
             \mod_moodleoverflow\readtracking::moodleoverflow_delete_read_records(-1, $post->id);
 
+            // Delete the attachments.
+            try {
```

Comment:
in this way you delete the db enrty, but NOT the actual file in the folder (db is just for where the file can be found) 
Change to somthing like 
```
$fs = get_file_storage();

// Prepare file record object
$fileinfo = array(
    'component' => 'mod_mymodule',
    'filearea' => 'myarea',     // usually = table name
    'itemid' => 0,               // usually = ID of row in table
    'contextid' => $context->id, // ID of context
    'filepath' => '/',           // any path beginning and ending in /
    'filename' => 'myfile.txt'); // any filename

// Get file
$file = $fs->get_file($fileinfo['contextid'], $fileinfo['component'], $fileinfo['filearea'], 
        $fileinfo['itemid'], $fileinfo['filepath'], $fileinfo['filename']);

// Delete it if it exists
if ($file) {
    $file->delete();
}
````
(https://docs.moodle.org/dev/File_API#Delete_file)